pkg/sysfs/system.go

```diff
@@ -1096,6 +1097,25 @@ func (c *cpu) GetLastLevelCaches() []*Cache {
 	return caches
 }
 
+// GetNthLevelCacheCPUSet returns the cpuset for the nth level caches of this CPU.
+func (c *cpu) GetNthLevelCacheCPUSet(n int) cpuset.CPUSet {
+	if len(c.caches) < 1 {
+		return c.ThreadCPUSet()
+	}
+
+	cpus := cpuset.New()
+
+	for _, cch := range c.caches {
+		if cch.level == n {
+			cpus = cpus.Union(CPUSetFromIDSet(cch.cpus))
+		} else if cch.level > n {
```

Comment:
No, `cpu.caches` are sorted, but they are sorted by index, not by level. This also guarantees that they are sorted by level as well. I can add a `if cch.level > level` breakout condition to terminate the loop early.

It is also true that `cpu.caches` are sorted because they are discovered from sysfs by globbing and sysfs returns them by lexiographic index order. So if there ever were more than 10 cache levels (I seriously doubt this would ever happen), then this breaks as lexiographic order != index order. I can add an explicit sort for them during discovery.

Reply:
Added explicit sorting of caches by index.